### PR TITLE
ENH: Re-enable a test for SSL encoders

### PR DESCRIPTION
**Pull request by ant0nsc**
Re-enable tests that had been disabled in #487. For that, remove the code duplication we had in getting an AML workspace specifically for unit tests. `get_workspace` now can read settings from environment variables.

### CODE REVIEW

File: hi-ml-azure/src/health_azure/utils.py [hunk 1]
```diff
@@ -55,7 +55,7 @@
 # name, hence version will always be fixed
 ENVIRONMENT_VERSION = "1"
 
-# Environment variables used for authentication
+# Environment variables used for authentication and workspace selection
```
Comment (kenza-bouzid) [inline hi-ml-azure/src/health_azure/utils.py:58]: Can we maybe break this comment? Specify which env variables are used for auth (59-61) and add a comment line 62 for Workspace selection env vars?

Comment (kenza-bouzid) [inline hi-ml-azure/src/health_azure/utils.py:58]: Can we maybe break this comment? Specify which env variables are used for auth (59-61) and add a comment line 62 for Workspace selection env vars?

File: hi-ml-azure/src/health_azure/utils.py [hunk 2]
```diff
@@ -721,21 +730,38 @@ def get_workspace(aml_workspace: Optional[Workspace] = None, workspace_config_pa
...
+    workspace_name = None
+    subscription_id = None
+    resource_group = None
+    if workspace_config_path is None:
+        logging.info("Trying to load the environment variables that define the workspace.")
+        workspace_name = get_secret_from_environment(ENV_WORKSPACE_NAME, allow_missing=True)
+        subscription_id = get_secret_from_environment(ENV_SUBSCRIPTION_ID, allow_missing=True)
+        resource_group = get_secret_from_environment(ENV_RESOURCE_GROUP, allow_missing=True)
+
+    if workspace_config_path is not None and not workspace_config_path.is_file():
```
Comment (kenza-bouzid) [inline hi-ml-azure/src/health_azure/utils.py:748]: [Minor] can we move this before  line 739, to keep the 5 steps isolated?

File: hi-ml-azure/src/health_azure/utils.py [hunk 3]
```diff
@@ -721,21 +730,38 @@ def get_workspace(aml_workspace: Optional[Workspace] = None, workspace_config_pa
...
+        raise FileNotFoundError(f"Workspace config file does not exist: {workspace_config_path}")
+
+    auth = get_authentication()
+    # Check if all 3 environment variables are set
+    if bool(workspace_name) and bool(subscription_id) and bool(resource_group):
```
Comment (kenza-bouzid) [inline hi-ml-azure/src/health_azure/utils.py:753]: [Minor] can we switch the order of 753 and 757 to keep the logic as described in doc string? is it meant to be sequential steps?

File: hi-ml-cpath/testhisto/testhisto/utils/test_deepmil_utils.py [hunk 4]
```diff
@@ -57,7 +56,6 @@ def test_load_ssl_checkpoint_from_url(tmp_path: Path) -> None:
     assert isinstance(encoder, SSLEncoder)
 
 
-@pytest.mark.skip(reason="This test is failing because of issue #655")
 def test_load_ssl_checkpoint_from_run_id(tmp_path: Path) -> None:
     encoder_params = EncoderParams(encoder_type=SSLEncoder.__name__, ssl_checkpoint=CheckpointParser(TEST_SSL_RUN_ID))
     assert encoder_params.ssl_checkpoint.is_aml_run_id
```
Comment (kenza-bouzid) [inline hi-ml-cpath/testhisto/testhisto/utils/test_deepmil_utils.py:61]: do we still need to patch get_workspace line 62? I saw that you removed it in test_encoder? will it be inferred from env variables of the github agent?

**Review by kenza-bouzid — CHANGES_REQUESTED** [review]
Thanks Anton for addressing this. My comments are mainly about reorganizing the code to match the description in doc string. Also there's still one test that need to be unskipped, it's test_load_slide in testhisto.preprocessing.test_loading

File: hi-ml-azure/src/health_azure/utils.py (hunk 3 above)
Comment (ant0nsc) [reply hi-ml-azure/src/health_azure/utils.py:753]: good spot, thanks!

File: hi-ml-azure/src/health_azure/utils.py (hunk 2 above)
Comment (ant0nsc) [reply hi-ml-azure/src/health_azure/utils.py:748]: Good suggestion, it reads nicer now.

File: hi-ml-azure/src/health_azure/utils.py (hunk 1 above)
Comment (ant0nsc) [reply hi-ml-azure/src/health_azure/utils.py:58]: Done

File: hi-ml-cpath/testhisto/testhisto/utils/test_deepmil_utils.py (hunk 4 above)
Comment (ant0nsc) [reply hi-ml-cpath/testhisto/testhisto/utils/test_deepmil_utils.py:61]: Very good point. Removed here and at several other places

File: hi-ml-azure/src/health_azure/utils.py [hunk 5]
```diff
@@ -699,19 +701,28 @@ def find_file_in_parent_to_pythonpath(file_name: str) -> Optional[Path]:
...
+      5. If 3 environment variables are found, use them to identify the workspace (`HIML_RESOURCE_GROUP`,
+        `HIML_SUBSCRIPTION_ID`, `HIML_WORKSPACE_NAME`)
+
+    If none of the above succeeds, an exception is raised.
 
     :param aml_workspace: If provided this is returned as the AzureML Workspace.
     :param workspace_config_path: If not provided with an AzureML Workspace, then load one given the information in this
         config
     :return: An AzureML workspace.
+    :raises: ValueError if neither of the available options for accessing the workspace succeeds.
```
Comment (mebristo) [inline hi-ml-azure/src/health_azure/utils.py:724]: Nit: If _none_ of the available options

Comment (ant0nsc) [reply hi-ml-azure/src/health_azure/utils.py]: resolve (plus the misplaced `:`)

**Review by mebristo — APPROVED** [review]

**Review by kenza-bouzid — APPROVED** [review]

**Pull request merged**